contracts/modules/factories/LineFactory.sol

```diff
@@ -7,62 +7,151 @@ import {LineFactoryLib} from "../../utils/LineFactoryLib.sol";
 import {SecuredLine} from "../credit/SecuredLine.sol";
 
 contract LineFactory is ILineFactory {
-
     IModuleFactory immutable factory;
-   
+
     uint8 constant defaultRevenueSplit = 90; // 90% to debt repayment
-    uint8 constant MAX_SPLIT = 100;          // max % to take
+    uint8 constant MAX_SPLIT = 100; // max % to take
     uint32 constant defaultMinCRatio = 3000; // 30.00% minimum collateral ratio
- 
-    constructor (address moduleFactory) {
+
+    address public immutable arbiter;
+    address public immutable oracle;
+    address public immutable swapTarget;
+
+    constructor(
+        address moduleFactory,
+        address arbiter_,
+        address oracle_,
+        address swapTarget_
+    ) {
         factory = IModuleFactory(moduleFactory);
-        
+        arbiter = arbiter_;
+        oracle = oracle_;
+        swapTarget = swapTarget_;
     }
-   
-    function deployEscrow(uint32 minCRatio, address oracle_, address owner, address borrower) external returns(address){
-        return factory.deployEscrow(minCRatio, oracle_, owner, borrower);
+
+    function deployEscrow(
+        uint32 minCRatio,
+        address owner,
+        address borrower
+    ) external returns (address) {
+        if (oracle == address(0)) {
```

Comment:
since we're setting oracle and arbiter as immutable state, We only need to check they aren't 0 in the constructor function once and then dont need to check again.